Security/How to Configure a Service Fabric Managed Cluster with Common Name Certificate.md

```diff
@@ -0,0 +1,259 @@
+3. Browse to certificate .pfx file to import.
+
+## Using common name for DNS resolution to connect to a managed cluster 
+
+In an unmanaged cluster, if configuring for use with a common name certificate, the 'managementEndpoint' can be modified to match the certificates SubjectName / common name for DNS resolution. For Service Fabric Managed  clusters, the managementEndpoint is not configurable. To resolve the name of the managed cluster using the certificates common name, an external configuration using DNS resolution is necessary, for example, by adding a CNAME record to DNS for the cluster FQDN address ({{cluster}}.{{location}}.cloudapp.azure.com).
+
+## PowerShell commands to get cluster server certificate thumbprint
+
+Run the commands below to use PowerShell to enumerate the managed cluster 'cluster' / server certificate.
+
+  ```powershell
+  $subscriptionId = ''
+  $resourceGroupName = ''
+  $managedClusterName = $resourceGroupName
+  $clusterResource = Get-AzResource -ResourceId "/subscriptions/$subscriptionId/resourcegroups/$resourceGroupName/providers/Microsoft.ServiceFabric/managedClusters/$managedClusterName"
+  $serverThumbprint = $clusterResource.Properties.clusterCertificateThumbprints
+  write-host "server thumbprint:$serverThumbprint"
+  ```
+
+## PowerShell command to connect to cluster
+
+Run the commands below to use PowerShell to connect to the managed cluster.
+
+```powershell
+$managementEndpoint = 'sfcluster.eastus.cloudapp.azure.com' # {{cluster name}}.{{location}}.cloudapp.azure.com
+$managementPort = 19000
+$serverThumbprint = ''
+$certificateCommonName = '*.sfcluster.com'
+Connect-ServiceFabricCluster -ConnectionEndpoint "$managementEndpoint`:$managementPort" `
+        -ServerCertThumbprint $serverThumbprint `
+        -StoreLocation CurrentUser `
+        -StoreName My `
+        -X509Credential `
+        -FindType FindBySubjectName `
+        -FindValue $certificateCommonName `
+        -Verbose
+```
+
+## Example PowerShell script
+
+The following script performs the commands documented in this TSG to connect to a Service Fabric Managed Cluster.
+
+```powershell
+invoke-webRequest "https://raw.githubusercontent.com/Azure/Service-Fabric-Troubleshooting-Guides/master/Scripts/sfm-connect.ps1" -outFile "$pwd\sfm-connect.ps1";
+.\sfm-connect.ps1 -clusterEndpoint sfcluster.eastus.cloudapp.azure.com -commonName *.sfcluster.com
+```
+
+## Troubleshooting
+
+- Verify the common name certificate is time valid and not revoked.
+
+- Verify there are not multiple certificates installed with same common name.
+
+- Verify the common name certificate being used is installed on machine where PowerShell commands are being executed. Opening Certmgr.msc on machine will open the 'CurrentUser' certificate store.
+
+- Verify the Issuer certificate thumbprint matches 'issuerThumbprint' in cluster 'clients' configuration for common name certificate. Renewed certificates may not have the same Issuer thumbprint.
```

Comment:
we should ask them to add all possible issuer TP's in cases where the CA may have multiple possible intermediaries issuing renewals.  Digicert does this

Reply:
updated